### PR TITLE
Add support for an S3 chunk store to TelstateDataSource

### PR DESCRIPTION
**Pull request by ludwigschwardt**
A quick one...

If the `s3_endpoint_url` key is found in telstate, prefer that to the RADOS option. The chunk_name is gone for now (since it should be simply the capture stream name), but it might make a comeback once the data source builds from a viewed-up telstate only.

### CODE REVIEW

File: katdal/datasources.py [hunk 1]
```diff
@@ -152,25 +154,27 @@ def __init__(self, telstate, capture_block_id=None, stream_name=None,
...
         else:
-            # Extract VisFlagsWeights and timestamps from telstate
-            with tempfile.NamedTemporaryFile() as f:
-                f.write(telstate['ceph_conf'])
-                f.flush()
-                pool = telstate['ceph_pool']
-                store = RadosChunkStore.from_config(f.name, pool)
-            ts_name = store.join(base_name, 'timestamps')
+            try:
+                store = S3ChunkStore.from_url(telstate['s3_endpoint_url'])
```
Comment (bmerry) [inline katdal/datasources.py]: You'll get a KeyError if `s3_endpoint_url` isn't there.

File: katdal/datasources.py [hunk 2]
```diff
@@ -152,25 +154,27 @@ def __init__(self, telstate, capture_block_id=None, stream_name=None,
...
+            except StoreUnavailable:
+                # Extract VisFlagsWeights and timestamps from telstate
```
Comment (bmerry) [inline katdal/datasources.py]: Should this comment be above the try? (since it applies to S3 as well)?

File: katdal/datasources.py (hunk 1 above)
Comment (ludwigschwardt) [reply katdal/datasources.py]: Fixed. S3 is also now the only option.

File: katdal/datasources.py (hunk 2 above)
Comment (ludwigschwardt) [reply katdal/datasources.py]: Done.

**Pull request merged**